panic on 500 from googleapi

```
time="2017-07-24T05:49:46Z" level=info msg="Starting stackdriver_exporter (version=0.1.1, branch=HEAD, revision=b4cf43ee4f11fcfef7effcc66ca7aa862219128d)" source="stackdriver_exporter.go:120"
time="2017-07-24T05:49:46Z" level=info msg="Build context (go=go1.8.3, user=root@1cc1402df604, date=20170715-22:22:05)" source="stackdriver_exporter.go:121"
time="2017-07-24T05:49:46Z" level=info msg="Listening on :9255" source="stackdriver_exporter.go:147"
time="2017-07-24T05:56:16Z" level=error msg="Error while getting Google Stackdriver Monitoring metrics: googleapi: Error 500: An internal error occurred., backendError" source="monitoring_collector.go:123"
time="2017-07-24T06:01:12Z" level=error msg="Error while getting Google Stackdriver Monitoring metrics: googleapi: Error 500: An internal error occurred., backendError" source="monitoring_collector.go:123"
panic: send on closed channel

goroutine 4421 [running]:
github.com/frodenas/stackdriver_exporter/collectors.(*MonitoringCollector).reportTimeSeriesMetrics(0xc42001d2c0, 0xc42023e070, 0xc4201e7110, 0xc4204c2420, 0xc42023e070, 0x0)
        /go/src/github.com/frodenas/stackdriver_exporter/collectors/monitoring_collector.go:298 +0xd6b
github.com/frodenas/stackdriver_exporter/collectors.(*MonitoringCollector).reportMonitoringMetrics.func1.1(0xc420320ac0, 0xc42001d2c0, 0xed10782fc, 0xc411225411, 0x0, 0xed1078428, 0xc41122549d, 0x0, 0xc420494f60, 0xc4201e7110, ...)
        /go/src/github.com/frodenas/stackdriver_exporter/collectors/monitoring_collector.go:174 +0x5ef
created by github.com/frodenas/stackdriver_exporter/collectors.(*MonitoringCollector).reportMonitoringMetrics.func1
        /go/src/github.com/frodenas/stackdriver_exporter/collectors/monitoring_collector.go:183 +0x28b
```

--- a/collectors/monitoring_collector.go
+++ b/collectors/monitoring_collector.go
@@ -145,8 +145,7 @@ func (c *MonitoringCollector) reportMonitoringMetrics(ch chan<- prometheus.Metri
 
 		c.apiCallsTotalMetric.Inc()
 
-		doneChannel := make(chan bool, 1)
-		errChannel := make(chan error, 1)
+		errChannel := make(chan error, len(page.MetricDescriptors))
 
 		startTime := time.Now().UTC().Add(c.metricsInterval * -1)
 		endTime := time.Now().UTC()
@@ -182,25 +181,15 @@ func (c *MonitoringCollector) reportMonitoringMetrics(ch chan<- prometheus.Metri
 				}
 			}(metricDescriptor, ch)
 		}
+		wg.Wait()
+		close(errChannel)
 
-		go func() {
-			wg.Wait()
-			close(doneChannel)
-		}()
-
-		select {
-		case <-doneChannel:
-		case err := <-errChannel:
-			return err
-		}
-
-		return nil
+		return <-errChannel
 	}
 
 	var wg = &sync.WaitGroup{}
 
-	doneChannel := make(chan bool, 1)
-	errChannel := make(chan error, 1)
+	errChannel := make(chan error, len(c.metricsTypePrefixes))
 
 	for _, metricsTypePrefix := range c.metricsTypePrefixes {
 		wg.Add(1)
@@ -216,18 +205,10 @@ func (c *MonitoringCollector) reportMonitoringMetrics(ch chan<- prometheus.Metri
 		}(metricsTypePrefix)
 	}
 
-	go func() {
-		wg.Wait()
-		close(doneChannel)
-	}()
+	wg.Wait()
+	close(errChannel)
 
-	select {
-	case <-doneChannel:
-	case err := <-errChannel:
-		return err
-	}
-
-	return nil
+	return <-errChannel
 }
 
 func (c *MonitoringCollector) reportTimeSeriesMetrics(page *monitoring.ListTimeSeriesResponse, metricDescriptor *monitoring.MetricDescriptor, ch chan<- prometheus.Metric) error {

